src/ctrl/nlas/mcast.rs

```diff
@@ -4,13 +4,42 @@ use crate::constants::*;
 use anyhow::Context;
 use byteorder::{ByteOrder, NativeEndian};
 use netlink_packet_utils::{
-    nla::{Nla, NlaBuffer},
+    nla::{Nla, NlaBuffer, NlasIterator, NLA_F_NESTED},
     parsers::*,
     traits::*,
     DecodeError,
 };
 use std::mem::size_of_val;
 
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct McastGroup {
+    pub nlas: Vec<McastGrpAttrs>,
+}
+
+impl Nla for McastGroup {
+    fn value_len(&self) -> usize {
+        self.nlas.iter().map(|nla| nla.buffer_len()).sum()
```

Comment:
Please use `self.nlas.as_slice().buffer_len()`

Reply:
Done, plus same for Op